### PR TITLE
Feature/qggpt 375 flesch

### PR DESCRIPTION
**Pull request by SalehiElla**
Added Readability Score

### CODE REVIEW

File: src/components/chat/chat-row.tsx [hunk 1]
```diff
@@ -148,11 +153,14 @@ export const ChatRow: FC<ChatRowProps> = props => {
           <Typography variant="h3" className="mt-0 flex-1 text-heading" tabIndex={0}>
             {props.name}
           </Typography>
-          {process.env.NODE_ENV === "development" && (
-            <Typography variant="h3" className="mt-0 flex-1 text-heading" tabIndex={0}>
-              {props.chatMessageId}
-            </Typography>
-          )}
+          <Typography variant="h3" className="mt-0 flex-1 text-center text-heading" tabIndex={0}>
```
Comment (QDAP-Fred) [inline src/components/chat/chat-row.tsx:156]: are we expecting to display to chatMessageId to everyone?

File: src/features/chat/chat-services/chat-flesch.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+function countWords(text: string): number {
+  return text.match(/\w+/g)?.length || 0
+}
+
+function countSentences(text: string): number {
+  return text.match(/[.!?]+(\s|$)/g)?.length || 0
+}
+
+export function calculateFleschScore(text: string): number {
+  const words = countWords(text)
```
Comment (QDAP-Fred) [inline src/features/chat/chat-services/chat-flesch.ts:12]: Safety first! wrap it in a try-catch

File: src/features/chat/models.ts [hunk 3]
```diff
@@ -67,13 +67,15 @@ export interface UserChatMessageModel extends ChatMessageModel {
...
   contentFilterResult?: unknown
+  userFleschScore?: number
 }
 export interface AssistantChatMessageModel extends ChatMessageModel {
   originalCompletion: string
   role: ChatRole.Assistant
   feedback: FeedbackType
   sentiment: ChatSentiment
   reason: string
+  systemFleshScore?: number
```
Comment (QDAP-Fred) [inline src/features/chat/models.ts:78]: use fleshScore property in ChatThreadModel

File: src/features/ui/assistant-buttons.tsx [hunk 4]
```diff
@@ -35,6 +34,48 @@ export const AssistantButtons: React.FC<AssistantButtonsProps> = ({
...
+            <span onClick={handleFleschIconClick} className={buttonClass}>
+              <BookOpenText size={iconSize} />
+            </span>
+          </Tooltip.Trigger>
+          <Tooltip.Content
+            side="top"
+            className="rounded-md bg-primary-foreground p-2 text-sm text-foreground shadow-lg"
+          >
+            <p>
+              <strong>Readability Level:</strong> This test rates text on a 100-point scale.<br></br> The higher the
```
Comment (QDAP-Fred) [inline src/features/ui/assistant-buttons.tsx:57]: fyi `<br></br>` can be simplified with `<br />`

**Review by QDAP-Fred — CHANGES_REQUESTED** [review]

File: src/components/chat/chat-row.tsx [hunk 5]
```diff
@@ -64,6 +65,8 @@ export const ChatRow: FC<ChatRowProps> = props => {
     }
   }
 
+  const handleFleschIconClick = () => {}
```
[the remark this answers is not in the document]
Comment (SalehiElla) [reply src/components/chat/chat-row.tsx]: Done

File: src/features/chat/chat-services/chat-flesch.ts (hunk 2 above)
Comment (SalehiElla) [reply src/features/chat/chat-services/chat-flesch.ts]: all sorted

File: src/features/ui/assistant-buttons.tsx (hunk 4 above)
Comment (SalehiElla) [reply src/features/ui/assistant-buttons.tsx]: Thanks, applied.

**Review by QDAP-Fred — APPROVED** [review]

**Pull request merged**